src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
In David's code, txns.append must return a reference. In the new code, it returns a *value*. `Object Array::appendObject`. We change the state of that value and then don't do anything else. I find that strange. I get there are things going on behind the scenes. The universal reference almost hides the fact that we're dealing with a function that returns a value (I understand why it's there, but it almost hides the fact that appendObject does not return a ref).